operator/internal/manifests/storage/configure.go

```diff
@@ -17,12 +18,12 @@ import (
 // - All: Ensure object storage secret mounted and auth projected as env vars.
 // - GCS: Ensure env var GOOGLE_APPLICATION_CREDENTIALS in container
 // - S3: Ensure mounting custom CA configmap if any TLSConfig given
-func ConfigureDeployment(d *appsv1.Deployment, opts Options) error {
+func ConfigureDeployment(d *appsv1.Deployment, opts Options, osEnabled bool) error {
```

Comment:
As mentioned above, if we pass the full `manifests.Options` struct here we don't need to reinvent naming for references to individual fields. In fact` osEnabled` does not immediately ring the bell for everyone not long acquainted with the code bases to `opts.FeatureGates.OpenShift.Enabled`, WDYT?